ipfscluster.go

```diff
@@ -176,6 +176,8 @@ type PeerMonitor interface {
 	// LatestMetrics returns a map with the latest metrics of matching name
 	// for the current cluster peers.
 	LatestMetrics(ctx context.Context, name string) []*api.Metric
+	// MetricNames returns a list of metric names.
+	MetricNames() []string
```

Comment:
Always pass contexts to interface methods, especially, when there a corresponding API endpoint, so that request cancellation propogates the entire way through the stack and also so it doesn't need to be added later for metrics/tracing stuff.

Reply:
thanks